test/replication-luatest/gh_9263_assertion_in_box_promote_qsync_test.lua

```diff
@@ -0,0 +1,153 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local proxy = require('luatest.replica_proxy')
+local server = require('luatest.server')
+
+local g = t.group('assertion-in-box-promote-qsync')
+
+local wait_timeout = 10
+
+local function wait_pair_sync(server1, server2)
+    -- Without retrying it fails sometimes when vclocks are empty and both
+    -- instances are in 'connect' state instead of 'follow'.
+    t.helpers.retrying({timeout = 10}, function()
+        server1:wait_for_vclock_of(server2)
+        server2:wait_for_vclock_of(server1)
+        server1:assert_follows_upstream(server2:get_instance_id())
+        server2:assert_follows_upstream(server1:get_instance_id())
+    end)
+end
+
+local function server_wait_wal_is_blocked(server)
+    server:exec(function()
+        t.helpers.retrying({timeout = 10}, function()
+            t.assert(box.error.injection.get('ERRINJ_WAL_DELAY'))
+        end)
+    end)
+end
+
+local function server_wait_synchro_queue_len_is_equal(server, expected)
+    server:exec(function(expected)
+        t.helpers.retrying({timeout = 10}, function(expected)
+            t.assert_equals(box.info.synchro.queue.len, expected)
+        end, expected)
+    end, {expected})
+end
+
+local function get_wait_quorum_count(server)
+    return server:exec(function()
+        return box.error.injection.get('ERRINJ_WAIT_QUORUM_COUNT')
+    end)
+end
+
+local function server_wait_wait_quorum_count_ge_than(server, threshold)
+    server:exec(function(threshold, wait_timeout)
+        t.helpers.retrying({timeout = wait_timeout}, function(threshold)
+            t.assert_ge(box.error.injection.get('ERRINJ_WAIT_QUORUM_COUNT'),
+                threshold)
+        end, threshold)
+    end, {threshold, wait_timeout})
+end
+
+g.before_each(function(cg)
+    t.tarantool.skip_if_not_debug()
+
+    cg.cluster = cluster:new({})
+
+    local box_cfg = {
+        replication = {
+            server.build_listen_uri('master', cg.cluster.id),
+            server.build_listen_uri('replica_proxy'),
+        },
+        election_mode = 'candidate',
+        replication_synchro_quorum = 2,
+        replication_synchro_timeout = 100000,
+        replication_timeout = 0.1,
+        election_fencing_mode='off',
+    }
+    box_cfg.instance_name='master'
+    cg.master = cg.cluster:build_and_add_server({
+        alias = 'master',
+        box_cfg = box_cfg
+    })
+    box_cfg.replication = {
+        server.build_listen_uri('replica', cg.cluster.id),
+        server.build_listen_uri('master_proxy'),
+    }
+    box_cfg.election_mode = 'off'
+    box_cfg.instance_name='replica'
```

Comment:
Lets be consistent and use `a = b` everywhere, not `a=b`.

Reply:
```diff
--- a/test/replication-luatest/gh_9263_assertion_in_box_promote_qsync_test.lua
+++ b/test/replication-luatest/gh_9263_assertion_in_box_promote_qsync_test.lua
@@ -63,9 +63,8 @@ g.before_each(function(cg)
         replication_synchro_quorum = 2,
         replication_synchro_timeout = 100000,
         replication_timeout = 0.1,
-        election_fencing_mode='off',
+        election_fencing_mode = 'off',
     }
-    box_cfg.instance_name='master'
     cg.master = cg.cluster:build_and_add_server({
         alias = 'master',
         box_cfg = box_cfg
@@ -75,7 +74,6 @@ g.before_each(function(cg)
         server.build_listen_uri('master_proxy'),
     }
     box_cfg.election_mode = 'off'
-    box_cfg.instance_name='replica'
     cg.replica = cg.cluster:build_and_add_server({
         alias = 'replica',
         box_cfg = box_cfg
@@ -141,8 +139,8 @@ g.test_is_in_box_promote = function(cg)
     server_wait_wait_quorum_count_ge_than(cg.replica, wait_quorum_count + 1)
     cg.replica:exec(function()
         box.cfg{
-            election_mode='candidate',
-            replication_synchro_quorum=1
+            election_mode = 'candidate',
+            replication_synchro_quorum = 1
         }
     end)
     cg.replica:wait_for_election_state('leader')
```